### PR TITLE
Prevent xss

### PR DESCRIPTION
**Pull request by tobi321**
[Trello Card](https://trello.com/c/5bz5FJ9G/616-xss-verhinden)

- Fixed the post folder test, with the correct name
- Changed the postSpecialCharsToFolderTest (ignoring the name of the folder) since we escape 'dangerous' characters due to XSS Prevention

### CODE REVIEW

**Review by wehrlean — CHANGES_REQUESTED** [review]
I had to adjust the classpath because it referred to Mac. Try it on your PC. 

The input has forbidden characters, like ; . This leads to errors, for example when you want to generate the model automatically. 
Possibilities that come to my mind:
1. You forward the error message to the frontend.
2. Remove all forbidden characters.
3. Bad for the user, because he wanted to smuggle in html.

Otherwise it works!

**Comment by tobi321** [comment]
Sorry for the inconvenience, the modified classpath works also on my machine. 

Valid point, but I am not sure if I get you correctly. 

When you enter the character `;` into the model generation input field, and it will be assigned to a variable, then you will get an error even without XSS prevention.

Consider the following scenario:
If the user enters a character which will be escaped (e.g. `&`) it will get escaped as `&amp;` which will show the user the message "forbidden character `;`". In this case, the user would remove the `;` and save it afterwards. At this stage the name would be `&amp`. However, after accessing the model again, the `&` will be escaped again and the user is back were he started. 

My approach would be to add the escaped characters `&  <  > "  '  / ` to the already define ones ( `, ; |`). 
 
Thoughts, @wehrlean?

**Comment by tobi321** [comment]
- Removed the history view, but kept the files. What do you think @wehrlean, should we delete them completely? 
- Changed the allowed pattern to allow every character

File: bundles/specmate-emfjson/src/com/specmate/emfjson/EMFJsonSerializer.java [hunk 1]
```diff
@@ -190,7 +191,8 @@ private Object serializeValue(Object value) throws SpecmateException {
 		} else if (value instanceof Boolean) {
 			return value;
 		} else {
-			return value.toString();
+			String v = value.toString();
+			return Encode.forHtml(v);
```
Comment (wehrlean) [inline bundles/specmate-emfjson/src/com/specmate/emfjson/EMFJsonSerializer.java]: If a node contains a '<' character, like _the age < 18_ the '<' is replaced as well. This shouldn't be

**Review by wehrlean — CHANGES_REQUESTED** [review]

**Pull request closed without merging**